R/Design-methods.R

```diff
@@ -1,165 +1,151 @@
+#' @include Data-methods.R
+#' @include Design-class.R
+#' @include McmcOptions-class.R
+#' @include Rules-methods.R
+#' @include Simulations-class.R
+#' @include helpers.R
+#' @include mcmc.R
+NULL
+
+# helper functions ----
+
+## set_seed ----
+
+#' Helper function to set and save the RNG seed
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' This code is basically copied from `stats:::simulate.lm`.
+#'
+#' @param seed an object specifying if and how the random number generator
+#' should be initialized (\dQuote{seeded}). Either `NULL` (default) or an
+#' integer that will be used in a call to [`set.seed`] before
+#' simulating the response vectors. If set, the value is saved as the
+#' `seed` slot of the returned object. The default, `NULL` will
+#' not change the random generator state.
+#' @return The RNGstate will be returned, in order to call this function
+#' with this input to reproduce the obtained simulation results.
+#'
+#' @export
+#' @keywords programming
+set_seed <- function(seed = NULL) {
   if (!exists(".Random.seed", envir = .GlobalEnv, inherits = FALSE)) {
     runif(1)
   }
 
   if (is.null(seed)) {
-    RNGstate <- get(".Random.seed", envir = .GlobalEnv)
+    rng_state <- get(".Random.seed", envir = .GlobalEnv)
   } else {
-    R.seed <- get(".Random.seed", envir = .GlobalEnv)
-    ## make sure R.seed exists in parent frame:
-    assign("R.seed", R.seed, envir = parent.frame())
+    r_seed <- get(".Random.seed", envir = .GlobalEnv)
+    # Make sure r_seed exists in parent frame.
+    assign("r_seed", r_seed, envir = parent.frame())
     set.seed(seed)
-    RNGstate <- structure(seed, kind = as.list(RNGkind()))
+    rng_state <- structure(seed, kind = as.list(RNGkind()))
     do.call("on.exit",
-      list(quote(assign(".Random.seed", R.seed, envir = .GlobalEnv))),
+      list(quote(assign(".Random.seed", r_seed, envir = .GlobalEnv))),
       envir = parent.frame()
     )
-    ## here we need the R.seed in the parent.frame!
+    # Here we need the r_seed in the parent.frame!
   }
 
-  return(RNGstate)
+  return(rng_state)
 }
 
+## get_result_list ----
+
+#' Helper function to obtain simulation results list
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' The function `fun` can use variables that are visible to itself.
+#' The names of these variables have to given in the vector `vars`.
+#'
+#' @param fun the simulation function for a single iteration, which takes as
+#' single parameter the iteration index.
+#' @param nsim number of simulations to be conducted.
+#' @param vars names of the variables.
+#' @param parallel should the simulation runs be parallelized across the
+#' clusters of the computer?
+#' @param n_cores how many cores should be used for parallel computing?
+#' @return The list with all simulation results (one iteration corresponds
+#' to one list element).
+#'
+#' @importFrom parallel makeCluster clusterEvalQ clusterExport clusterCall parLapply stopCluster
```

Comment:
this is a bit of a change, now we only want to import a single function from a package, and we always use `parallel::makeCluster()` i.e. with the namespace specification.
```suggestion
#' @importFrom parallel makeCluster 
```

Reply:
so we just need a single import of something, it does not have to be `makeCluster` though, to make `R CMD check` happy